docs/source/en/model_doc/superglue.md

```diff
@@ -0,0 +1,148 @@
+pairs. SuperGlue outperforms other learned approaches and achieves state-of-the-art results on the task of pose estimation in 
+challenging real-world indoor and outdoor environments. The proposed method performs matching in real-time on a modern GPU and 
+can be readily integrated into modern SfM or SLAM systems. The code and trained weights are publicly available at this [URL](https://github.com/magicleap/SuperGluePretrainedNetwork).*
+
+## How to use
+
+Here is a quick example of using the model. Since this model is an image matching model, it requires pairs of images to be matched:
+
+```python
+from transformers import AutoImageProcessor, AutoModel
+import torch
+from PIL import Image
+import requests
+
+url = "https://github.com/magicleap/SuperGluePretrainedNetwork/blob/master/assets/phototourism_sample_images/london_bridge_78916675_4568141288.jpg?raw=true"
+im1 = Image.open(requests.get(url, stream=True).raw)
+url = "https://github.com/magicleap/SuperGluePretrainedNetwork/blob/master/assets/phototourism_sample_images/london_bridge_19481797_2295892421.jpg?raw=true"
+im2 = Image.open(requests.get(url, stream=True).raw)
+images = [im1, im2]
+
+processor = AutoImageProcessor.from_pretrained("stevenbucaille/superglue_outdoor")
+model = AutoModel.from_pretrained("stevenbucaille/superglue_outdoor")
+
+inputs = processor(images, return_tensors="pt")
+outputs = model(**inputs)
+```
+
+The outputs contain the list of keypoints detected by the keypoint detector as well as the list of matches with their corresponding matching scores.
+Due to the nature of SuperGlue, to output a dynamic number of matches, you will need to use the mask attribute to retrieve the respective information :
+
+```python
+from transformers import AutoImageProcessor, AutoModel
+import torch
+from PIL import Image
+import requests
+
+url_image_1 = "http://images.cocodataset.org/val2017/000000039769.jpg"
+image_1 = Image.open(requests.get(url_image_1, stream=True).raw)
+url_image_2 = "http://images.cocodataset.org/test-stuff2017/000000000568.jpg"
+image_2 = Image.open(requests.get(url_image_2, stream=True).raw)
+
+images = [image_1, image_2]
+
+processor = AutoImageProcessor.from_pretrained("stevenbucaille/superglue_outdoor")
+model = AutoModel.from_pretrained("stevenbucaille/superglue_outdoor")
+
+inputs = processor(images, return_tensors="pt")
+outputs = model(**inputs)
+
+for i in len(images) // 2:
+    image0_mask = outputs.mask[i, 0]
+    image0_indices = torch.nonzero(image0_mask).squeeze()
+    image0_matches = outputs.matches[i, 0][image0_indices]
+    image0_matching_scores = outputs.matching_scores[i, 0][image0_indices]
+    image1_mask = outputs.mask[i, 1]
+    image1_indices = torch.nonzero(image1_mask).squeeze()
+    image1_matches = outputs.matches[i, 1][image1_indices]
+    image1_matching_scores = outputs.matching_scores[i, 1][image1_indices]
```

Comment:
The mask returned by the model is, as mentioned above, of the shape `[batch_size, 2, max_keypoints]` where `max_keypoints` is the maximum number of keypoints detected in all images fed into the model by SuperPoint. In order to have only one mask for the entire inference, no matter the number of keypoints detected, we use the same mask batching mechanism we used in SuperPoint which informs the user what values in `matches` and `matching_scores` correspond to a specific image.
We could only use matches as indices like the following code : 
```python
image0_matches = outputs.matches[0, 0]
image0_matches = image0_matches[image0_matches != -1]
image1_keypoints = outputs.keypoints[0, 1][image0_matches]
```
Since image0_matches returns either indices or -1 in case of no match, we can indeed filter like above to retrieve matched keypoints in the other image and not use mask at all. 
The downside is the following : for an image with `n` keypoints detected with `n < max_keypoints`, then keypoints values for indices > `n` will have value = [0, 0] because of the batching, so the user won't be able to retrieve SuperPoint data without the mask, unless he filters out the [0, 0] values. But then it becomes invalid in the case where the `border_removal_distance` argument in SuperPointConfig is equal to 0, then it would be possible for SuperPoint to return a keypoint value of [0, 0]